cmd/podman/inspect.go

```diff
@@ -71,81 +71,108 @@ func inspectCmd(c *cli.Context) error {
 	if !libpod.StringInSlice(inspectType, []string{inspectTypeContainer, inspectTypeImage, inspectAll}) {
 		return errors.Errorf("the only recognized types are %q, %q, and %q", inspectTypeContainer, inspectTypeImage, inspectAll)
 	}
-	if !latestContainer {
-		name = args[0]
-	}
-	if latestContainer {
-		inspectType = inspectTypeContainer
-	}
+
 	outputFormat := c.String("format")
 	if strings.Contains(outputFormat, "{{.Id}}") {
 		outputFormat = strings.Replace(outputFormat, "{{.Id}}", formats.IDString, -1)
 	}
-
-	var data interface{}
-	switch inspectType {
-	case inspectTypeContainer:
-		var ctr *libpod.Container
-		var err error
-		if latestContainer {
-			ctr, err = runtime.GetLatestContainer()
-		} else {
-			ctr, err = runtime.LookupContainer(name)
-		}
-		if err != nil {
-			return errors.Wrapf(err, "error looking up container %q", name)
-		}
-		libpodInspectData, err := ctr.Inspect(c.Bool("size"))
-		if err != nil {
-			return errors.Wrapf(err, "error getting libpod container inspect data %q", ctr.ID)
-		}
-		data, err = getCtrInspectInfo(ctr, libpodInspectData)
-		if err != nil {
-			return errors.Wrapf(err, "error parsing container data %q", ctr.ID())
-		}
-	case inspectTypeImage:
-		image, err := runtime.GetImage(name)
-		if err != nil {
-			return errors.Wrapf(err, "error getting image %q", name)
-		}
-		data, err = runtime.GetImageInspectInfo(*image)
+	if latestContainer {
+		lc, err := runtime.GetLatestContainer()
 		if err != nil {
-			return errors.Wrapf(err, "error parsing image data %q", image.ID)
-		}
-	case inspectAll:
-		ctr, err := runtime.LookupContainer(name)
-		if err != nil {
-			image, err := runtime.GetImage(name)
-			if err != nil {
-				return errors.Wrapf(err, "error getting image %q", name)
-			}
-			data, err = runtime.GetImageInspectInfo(*image)
-			if err != nil {
-				return errors.Wrapf(err, "error parsing image data %q", image.ID)
-			}
-		} else {
-			libpodInspectData, err := ctr.Inspect(c.Bool("size"))
-			if err != nil {
-				return errors.Wrapf(err, "error getting libpod container inspect data %q", ctr.ID)
-			}
-			data, err = getCtrInspectInfo(ctr, libpodInspectData)
-			if err != nil {
-				return errors.Wrapf(err, "error parsing container data %q", ctr.ID)
-			}
+			return err
 		}
+		args = append(args, lc.ID())
+		inspectType = inspectTypeContainer
+	}
+	inspectedObjects, iterateErr := iterateInput(c, args, runtime, inspectType)
+	if err != nil {
+
```

Comment:
Should there be a "return err" here?

Reply:
actually that is left over from me trying to make sure we gracefully fail when inspecting a list of things.